main.go

```diff
@@ -14,22 +16,52 @@ type Client struct {
 	BindUser     string
 	BindPassword string
 	TLS          bool
+	TLSCACert    string
+	TLSInsecure  bool
 }
 
+// Connect Creates un ldap connection
 func (c *Client) Connect() error {
-
+	var ldapConn *ldap.Conn
 	uri := fmt.Sprintf("%s:%d", c.Host, c.Port)
 
-	l, err := ldap.Dial("tcp", uri)
-	if err != nil {
-		return fmt.Errorf("error dialing: %s", err)
+	if c.TLS {
+		// Get the System Cert Pool
+		caCertPool, err := x509.SystemCertPool()
+		if err != nil {
+			return fmt.Errorf("error tls: %s", err)
+		}
+
+		// Use the provided CA certificate
+		if c.TLSCACert != "" {
+			caCertPool = x509.NewCertPool()
+			if ok := caCertPool.AppendCertsFromPEM([]byte(c.TLSCACert)); !ok {
+				return fmt.Errorf("error tls: Can't add the CA certificate to certificate pool")
+			}
+		}
+
+		conn, err := ldap.DialTLS("tcp", uri, &tls.Config{
+			RootCAs:            caCertPool,
+			InsecureSkipVerify: c.TLSInsecure,
+		})
+		if err != nil {
+			return fmt.Errorf("error dialing: %s", err)
+		}
+		ldapConn = conn
```

Comment:
I think we can simplify this a bit by using directly "c.Conn" to store generated connection. Same for lines 57 and 60. So we can remove line 64 too. Apart from that LGTM :+1:

Reply:
Fixed with https://github.com/Ouest-France/goldap/pull/3/commits/54a6a8f1808e687aff6e8650cca1738d940ca0fd